### PR TITLE
syscall: add task and boardctl APIs to syscall

### PR DESCRIPTION
**Pull request by jeongchanKim**
1. boardctl : This API can control the board directly such as power off. User can call this API.
2. task_setcancelstate, task_setcanceltype, task_testcancel : These APIs can set the cancel information in the tcb.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [3fee2180a958e8693bb6f5548a2bd223be5e50ea] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [3fee2180a958e8693bb6f5548a2bd223be5e50ea] - Code Rule Check OK.

File: os/syscall/syscall_lookup.h [hunk 1]
```diff
@@ -102,8 +102,14 @@ SYSCALL_LOOKUP(pgalloc,                   2, STUB_pgalloc)
 #endif
 SYSCALL_LOOKUP(task_delete,               1, STUB_task_delete)
 SYSCALL_LOOKUP(task_restart,              1, STUB_task_restart)
+SYSCALL_LOOKUP(task_setcancelstate,       2, STUB_task_setcancelstate)
+SYSCALL_LOOKUP(task_testcancel,           0, STUB_task_testcancel)
 SYSCALL_LOOKUP(up_assert,                 2, STUB_up_assert)
 
+#  ifdef CONFIG_CANCELLATION_POINTS
```
Comment (YiHyunJin) [inline os/syscall/syscall_lookup.h]: I think syscall_lookup.h and syscall.h are must same config. but syscall.h have not CONFIG_CANCELLATION_POINTS.

File: os/syscall/syscall_lookup.h [hunk 2]
```diff
@@ -102,8 +102,14 @@ SYSCALL_LOOKUP(pgalloc,                   2, STUB_pgalloc)
...
+SYSCALL_LOOKUP(task_setcanceltype,        2, STUB_task_setcanceltype)
```
Comment (YiHyunJin) [inline os/syscall/syscall_lookup.h]: I think the order of SYSCALL_LOOKUP and define 'syscall.h' should be the same.

Comment (jeongchanKim) [reply os/syscall/syscall_lookup.h]: The order is okay, but I will change for readability.

**Pull request merged**